pkg/scheduler/internal/cache/comparer/comparer_test.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package factory
+package comparer
```

Comment:
If it's trivial, please change package name to `comparer_test` in this PR.

Reply:
Why do we should name the package as `combiner_test`? Sorry, I can't understand the benefits of doing this?

But, even if we do, what should we name our file name? If we don't make any changes of files in combiner_test, we will get:
`pkg/scheduler/factory/factory.go:63:2: found packages comparer (comparer.go) and comparer (comparer_test.go) `

(I'm sure I have set the package name in comparer.go and  comparer_test.go as comparer_test. Perhaps contrary to Go's naming rules?)

Reply:
I think this is coming from my c++ background, where we try to test via public interfaces. For example, suppose one of the structs in our package has a private field protected by a private `sync.Mutex`. Nothing prevents a unit test from accessing the field without holding the mutex. (Hopefully `go test -race` would catch it.)

The point is that direct access to private fields might lead us to test the wrong thing. To follow from the previous example, if we have a strong guarantee that all access to the field is protected by a mutex, we can make certain assumptions about concurrent access.

Regardless of the package name, the file name "comparer_test.go" is appropriate.

/lgtm